### PR TITLE
API errrors, how to handle

### LINKED ISSUE #20949: UseExceptionHandler vs GlobalExceptionMiddleware
I read this article but what I am not understanding is that while there is Middleware Exception Handling in place since the early stages of .Net Core which allows us to implement a custom Middleware for capturing all exceptions and logging them without requiring a controller [(see here)](https://docs.microsoft.com/en-us/archive/blogs/brandonh/using-middleware-to-trap-exceptions-in-asp-net-core?source=docs) why one should go with UseExceptionHandler("/error") solution that involves the implementation of an extra controller (called ErrorController) to handle exceptions. As you know this will take the opportunity of having a common library for exception handling across all projects (Web APIs) in a product. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3f6504cd-a059-c5d4-73fb-61222ef1c6e0
* Version Independent ID: 1b42352c-a34f-7703-aa3e-d7e3156bd0bd
* Content: [Handle errors in ASP.NET Core web APIs](https://docs.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-5.0#exception-handler)
* Content Source: [aspnetcore/web-api/handle-errors.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/web-api/handle-errors.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-webapi**
* GitHub Login: @pranavkm
* Microsoft Alias: **prkrishn**

### PR DESCRIPTION
**Pull request by aehomay**
Fixes #20949

### CODE REVIEW

File: aspnetcore/web-api/handle-errors.md [hunk 1]
```diff
@@ -320,3 +320,8 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
 [!code-csharp[](index/samples/2.x/2.2/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=9-10)]
 
 ::: moniker-end
+
+## Use a Custome Middleware to handle exceptions
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/handle-errors.md:324]:
```suggestion
## Use a Custom Middleware to handle exceptions
```

File: aspnetcore/web-api/handle-errors.md [hunk 2]
```diff
@@ -320,3 +320,7 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
...
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/handle-errors.md]:
```suggestion
## Use a Custom Middleware to handle exceptions
```

File: aspnetcore/web-api/handle-errors.md [hunk 3]
```diff
@@ -320,3 +320,7 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
 [!code-csharp[](index/samples/2.x/2.2/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=9-10)]
 
 ::: moniker-end
+
+## Use a Custom Middleware to handle exceptions
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/handle-errors.md:324]:
```suggestion
## Custom Middleware to handle exceptions
```

File: aspnetcore/web-api/handle-errors.md [hunk 4]
```diff
@@ -320,3 +320,7 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
 [!code-csharp[](index/samples/2.x/2.2/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=9-10)]
 
 ::: moniker-end
+
+## Custom Middleware to handle exceptions
+
+While the built-in Exception Handling Middleware of ASP.NET Core is providing a convenient solution to handle exceptions in the chain of .NET Core layers, there will be, some cases in which having a common and customized Exception Handling Middleware will be a more elegant solution. For example, consider a project that is going to consume several fine-grained Web APIs. These APIs are all based on .NET Core and they all need to handle exceptions in more less similar way and log them by relyong on [ILogger](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0). For such a scenario having a Customized Middleware Exception Handling in a shared library among all APIs would be a good solution to increase code useability and provide a centralized logging and exception handling mechanism. Follwoing you can find a simple excample of Customized Exception Handling Middleware.
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/handle-errors.md:326]: The built in exception handling Middleware works well for most apps. For apps that require specialized exception handling, consider writing a custom exception handling Middleware. For example, library authors my prefer to write a custom middleware to handle errors in library APIs.

Comment (Rick-Anderson) [reply aspnetcore/web-api/handle-errors.md:326]:
```suggestion
The built in exception handling Middleware works well for most apps. For apps that require specialized exception handling, consider writing a custom exception handling Middleware. For example, library authors my prefer to write a custom middleware to handle errors in library APIs.
```

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/web-api/handle-errors.md [hunk 5]
```diff
@@ -320,3 +320,7 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
 [!code-csharp[](index/samples/2.x/2.2/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=9-10)]
 
 ::: moniker-end
+
+## Custom Middleware to handle exceptions
+
+The built in exception handling Middleware works well for most apps. For apps that require specialized exception handling, consider writing a custom exception handling Middleware. For example, library authors my prefer to write a custom middleware to handle errors in library APIs.
```
Comment (pranavkm) [inline aspnetcore/web-api/handle-errors.md:326]:
```suggestion
The defaults in the exception handling middleware works well for most apps. For apps that require specialized exception handling, consider [customizing the exception handling middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0#exception-handler-lambda).
```

File: aspnetcore/web-api/handle-errors.md [hunk 6]
```diff
@@ -320,3 +320,7 @@ Use the <xref:Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.ClientErrorMapping%2A>
 [!code-csharp[](index/samples/2.x/2.2/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=9-10)]
 
 ::: moniker-end
+
+## Custom Middleware to handle exceptions
+
+The defaults in the exception handling middleware works well for most apps. For apps that require specialized exception handling, consider [customizing the exception handling middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0#exception-handler-lambda).
```
Comment (Rick-Anderson) [inline aspnetcore/web-api/handle-errors.md:326]:
```suggestion
The defaults in the exception handling middleware works well for most apps. For apps that require specialized exception handling, consider [customizing the exception handling middleware](xref:fundamentals/error-handling#exception-handler-lambda).
```

**Comment by Rick-Anderson** [comment]
@pranavkm one minute review please.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**